sdk/core/core-http/src/operationSpec.ts

```diff
@@ -43,6 +43,19 @@ export interface OperationSpec {
    */
   readonly contentType?: string;
 
+  /**
+   * The media type of the request body.
+   * This value can be used to aide in serialization if it is provided.
+   */
+  readonly mediaType?:
+    | "json"
+    | "xml"
+    | "form"
+    | "binary"
+    | "multipart"
+    | "text"
+    | "unknown"
+    | string;
```

Comment:
@bterlson is it worth using your TS magic so this type doesn't collapse to `string`? 

Also, should we pull the type itself out into its own export for easy use externally?

Reply:
Alternatively, we could make this simply `string` and have some known enum of types?

Reply:
@xirzec Is it safe to resolve this comment based on the other thread, or do you still see value in creating a type that encompasses these known string values?

My motivation for in-lining the union rather than exporting another type was I only expect autorest to use this so I didn't want to pollute our exports but I can add it if you think it'd be valuable!

Reply:
I'm fine with this as-is